File: EF.RPC.Sharing/src/main/java/com/wiqer/rpc/sharing/core/RpcResponse.java
package com.wiqer.rpc.sharing.core;

/**
 * RPC响应消息
 * 遵循统一架构设计规范，定义标准化的RPC响应格式
 * 
 * @author EFRPC Team
 * @version 1.0.0
 * @since 2024-01-01
 */
public class RpcResponse extends RpcMessage {
    
    /**
     * 请求消息ID
     */
    private String requestId;
    
    /**
     * 调用结果
     */
    private Object result;
    
    /**
     * 异常信息
     */
    private Throwable exception;
    
    /**
     * 响应状态码
     */
    private int statusCode;
    
    /**
     * 响应状态消息
     */
    private String statusMessage;
    
    /**
     * 构造函数
     */
    public RpcResponse() {
        super();
        setMessageType(MessageType.RESPONSE);
        this.statusCode = 200;
        this.statusMessage = "OK";
    }
    
    /**
     * 构造函数
     * 
     * @param messageId 消息ID
     * @param requestId 请求消息ID
     */
    public RpcResponse(String messageId, String requestId) {
        super(messageId, MessageType.RESPONSE);
        this.requestId = requestId;
        this.statusCode = 200;
        this.statusMessage = "OK";
    }
    
    /**
     * 构造函数
     * 
     * @param requestId 请求消息ID
     * @param result 调用结果
     */
    public RpcResponse(String requestId, Object result) {
        super();
        setMessageType(MessageType.RESPONSE);
        this.requestId = requestId;
        this.result = result;
        this.statusCode = 200;
        this.statusMessage = "OK";
    }
    
    /**
     * 构造函数
     * 
     * @param requestId 请求消息ID
     * @param exception 异常信息
     */
    public RpcResponse(String requestId, Throwable exception) {
        super();
        setMessageType(MessageType.RESPONSE);
        this.requestId = requestId;
        this.exception = exception;
        this.statusCode = 500;
        this.statusMessage = "Internal Server Error";
    }
    
    /**
     * 获取请求消息ID
     * 
     * @return 请求消息ID
     */
    public String getRequestId() {
        return requestId;
    }
    
    /**
     * 设置请求消息ID
     * 
     * @param requestId 请求消息ID
     */
    public void setRequestId(String requestId) {
        this.requestId = requestId;
    }
    
    /**
     * 获取调用结果
     * 
     * @return 调用结果
     */
    public Object getResult() {
        return result;
    }
    
    /**
     * 设置调用结果
     * 
     * @param result 调用结果
     */
    public void setResult(Object result) {
        this.result = result;
    }
    
    /**
     * 获取异常信息
     * 
     * @return 异常信息
     */
    public Throwable getException() {
        return exception;
    }
    
    /**
     * 设置异常信息
     * 
     * @param exception 异常信息
     */
    public void setException(Throwable exception) {
        this.exception = exception;
        if (exception != null) {
            this.statusCode = 500;
            this.statusMessage = "Internal Server Error";
        }
    }
    
    /**
     * 获取响应状态码
     * 
     * @return 响应状态码
     */
    public int getStatusCode() {
        return statusCode;
    }
    
    /**
     * 设置响应状态码
     * 
     * @param statusCode 响应状态码
     */
    public void setStatusCode(int statusCode) {
        this.statusCode = statusCode;
    }
    
    /**
     * 获取响应状态消息
     * 
     * @return 响应状态消息
     */
    public String getStatusMessage() {
        return statusMessage;
    }
    
    /**
     * 设置响应状态消息
     * 
     * @param statusMessage 响应状态消息
     */
    public void setStatusMessage(String statusMessage) {
        this.statusMessage = statusMessage;
    }
    
    /**
     * 检查是否有异常
     * 
     * @return true表示有异常，false表示无异常
     */
    public boolean hasException() {
        return exception != null;
    }
    
    /**
     * 检查是否成功
     * 
     * @return true表示成功，false表示失败
     */
    public boolean isSuccess() {
        return !hasException() && statusCode >= 200 && statusCode < 300;
    }
    
    /**
     * 检查是否超时
     * 
     * @return true表示超时，false表示未超时
     */
    public boolean isTimeout() {
        return statusCode == 408;
    }
    
    /**
     * 检查是否服务不可用
     * 
     * @return true表示服务不可用，false表示服务可用
     */
    public boolean isServiceUnavailable() {
        return statusCode == 503;
    }
    
    /**
     * 创建成功响应
     * 
     * @param requestId 请求消息ID
     * @param result 调用结果
     * @return RPC响应
     */
    public static RpcResponse success(String requestId, Object result) {
        return new RpcResponse(requestId, result);
    }
    
    /**
     * 创建失败响应
     * 
     * @param requestId 请求消息ID
     * @param exception 异常信息
     * @return RPC响应
     */
    public static RpcResponse failure(String requestId, Throwable exception) {
        return new RpcResponse(requestId, exception);
    }
    
    /**
     * 创建超时响应
     * 
     * @param requestId 请求消息ID
     * @return RPC响应
     */
    public static RpcResponse timeout(String requestId) {
        RpcResponse response = new RpcResponse();
        response.setRequestId(requestId);
        response.setStatusCode(408);
        response.setStatusMessage("Request Timeout");
        return response;
    }
    
    /**
     * 创建服务不可用响应
     * 
     * @param requestId 请求消息ID
     * @return RPC响应
     */
    public static RpcResponse serviceUnavailable(String requestId) {
        RpcResponse response = new RpcResponse();
        response.setRequestId(requestId);
        response.setStatusCode(503);
        response.setStatusMessage("Service Unavailable");
        return response;
    }
    
    @Override
    public String toString() {
        return String.format("RpcResponse{messageId='%s', requestId='%s', success=%s, statusCode=%d, timestamp=%d}",
                getMessageId(), requestId, isSuccess(), statusCode, getTimestamp());
    }
    
    @Override
    public boolean equals(Object obj) {
        if (this == obj) return true;
        if (obj == null || getClass() != obj.getClass()) return false;
        if (!super.equals(obj)) return false;
        
        RpcResponse that = (RpcResponse) obj;
        
        if (statusCode != that.statusCode) return false;
        if (requestId != null ? !requestId.equals(that.requestId) : that.requestId != null) return false;
        if (result != null ? !result.equals(that.result) : that.result != null) return false;
        if (exception != null ? !exception.equals(that.exception) : that.exception != null) return false;
        return statusMessage != null ? statusMessage.equals(that.statusMessage) : that.statusMessage == null;
    }
    
    @Override
    public int hashCode() {
        int result = super.hashCode();
        result = 31 * result + (requestId != null ? requestId.hashCode() : 0);
        result = 31 * result + (this.result != null ? this.result.hashCode() : 0);
        result = 31 * result + (exception != null ? exception.hashCode() : 0);
        result = 31 * result + statusCode;
        result = 31 * result + (statusMessage != null ? statusMessage.hashCode() : 0);
        return result;
    }
    
    /**
     * 创建RPC响应的构建器
     * 
     * @return 响应构建器
     */
    public static Builder builder() {
        return new Builder();
    }
    
    /**
     * RPC响应构建器
     */
    public static class Builder {
        private String messageId;
        private String requestId;
        private Object result;
        private Throwable exception;
        private int statusCode = 200;
        private String statusMessage = "OK";
        
        public Builder messageId(String messageId) {
            this.messageId = messageId;
            return this;
        }
        
        public Builder requestId(String requestId) {
            this.requestId = requestId;
            return this;
        }
        
        public Builder result(Object result) {
            this.result = result;
            return this;
        }
        
        public Builder exception(Throwable exception) {
            this.exception = exception;
            if (exception != null) {
                this.statusCode = 500;
                this.statusMessage = "Internal Server Error";
            }
            return this;
        }
        
        public Builder statusCode(int statusCode) {
            this.statusCode = statusCode;
            return this;
        }
        
        public Builder statusMessage(String statusMessage) {
            this.statusMessage = statusMessage;
            return this;
        }
        
        public RpcResponse build() {
            RpcResponse response = new RpcResponse(messageId, requestId);
            response.setResult(result);
            response.setException(exception);
            response.setStatusCode(statusCode);
            response.setStatusMessage(statusMessage);
            return response;
        }
    }
} 

File: external-dependencies/MQTT-RPC.NET/EF.RPC.Sharing/RpcResponse.cs
using System;

namespace EF.RPC.Sharing
{
    /// <summary>
    /// RPC响应消息
    /// 遵循统一架构设计规范，定义标准化的RPC响应格式
    /// </summary>
    public class RpcResponse : RpcMessage
    {
        /// <summary>
        /// 请求消息ID
        /// </summary>
        public string RequestId { get; set; }
        
        /// <summary>
        /// 调用结果
        /// </summary>
        public object Result { get; set; }
        
        /// <summary>
        /// 异常信息
        /// </summary>
        public Exception Exception { get; set; }
        
        /// <summary>
        /// 响应状态码
        /// </summary>
        public int StatusCode { get; set; }
        
        /// <summary>
        /// 响应状态消息
        /// </summary>
        public string StatusMessage { get; set; }
        
        /// <summary>
        /// 构造函数
        /// </summary>
        public RpcResponse() : base()
        {
            MessageType = MessageType.Response;
            StatusCode = 200;
            StatusMessage = "OK";
        }
        
        /// <summary>
        /// 构造函数
        /// </summary>
        /// <param name="messageId">消息ID</param>
        /// <param name="requestId">请求消息ID</param>
        public RpcResponse(string messageId, string requestId) 
            : base(messageId, MessageType.Response)
        {
            RequestId = requestId;
            StatusCode = 200;
            StatusMessage = "OK";
        }
        
        /// <summary>
        /// 构造函数
        /// </summary>
        /// <param name="requestId">请求消息ID</param>
        /// <param name="result">调用结果</param>
        public RpcResponse(string requestId, object result) 
            : base()
        {
            MessageType = MessageType.Response;
            RequestId = requestId;
            Result = result;
            StatusCode = 200;
            StatusMessage = "OK";
        }
        
        /// <summary>
        /// 构造函数
        /// </summary>
        /// <param name="requestId">请求消息ID</param>
        /// <param name="exception">异常信息</param>
        public RpcResponse(string requestId, Exception exception) 
            : base()
        {
            MessageType = MessageType.Response;
            RequestId = requestId;
            Exception = exception;
            StatusCode = 500;
            StatusMessage = "Internal Server Error";
        }
        
        /// <summary>
        /// 检查是否有异常
        /// </summary>
        /// <returns>true表示有异常，false表示无异常</returns>
        public bool HasException()
        {
            return Exception != null;
        }
        
        /// <summary>
        /// 检查是否成功
        /// </summary>
        /// <returns>true表示成功，false表示失败</returns>
        public bool IsSuccess()
        {
            return !HasException() && StatusCode >= 200 && StatusCode < 300;
        }
        
        /// <summary>
        /// 检查是否超时
        /// </summary>
        /// <returns>true表示超时，false表示未超时</returns>
        public bool IsTimeout()
        {
            return StatusCode == 408;
        }
        
        /// <summary>
        /// 检查是否服务不可用
        /// </summary>
        /// <returns>true表示服务不可用，false表示服务可用</returns>
        public bool IsServiceUnavailable()
        {
            return StatusCode == 503;
        }
        
        /// <summary>
        /// 创建成功响应
        /// </summary>
        /// <param name="requestId">请求消息ID</param>
        /// <param name="result">调用结果</param>
        /// <returns>RPC响应</returns>
        public static RpcResponse Success(string requestId, object result)
        {
            return new RpcResponse(requestId, result);
        }
        
        /// <summary>
        /// 创建失败响应
        /// </summary>
        /// <param name="requestId">请求消息ID</param>
        /// <param name="exception">异常信息</param>
        /// <returns>RPC响应</returns>
        public static RpcResponse Failure(string requestId, Exception exception)
        {
            return new RpcResponse(requestId, exception);
        }
        
        /// <summary>
        /// 创建超时响应
        /// </summary>
        /// <param name="requestId">请求消息ID</param>
        /// <returns>RPC响应</returns>
        public static RpcResponse Timeout(string requestId)
        {
            var response = new RpcResponse();
            response.RequestId = requestId;
            response.StatusCode = 408;
            response.StatusMessage = "Request Timeout";
            return response;
        }
        
        /// <summary>
        /// 创建服务不可用响应
        /// </summary>
        /// <param name="requestId">请求消息ID</param>
        /// <returns>RPC响应</returns>
        public static RpcResponse ServiceUnavailable(string requestId)
        {
            var response = new RpcResponse();
            response.RequestId = requestId;
            response.StatusCode = 503;
            response.StatusMessage = "Service Unavailable";
            return response;
        }
        
        /// <summary>
        /// 重写ToString方法
        /// </summary>
        /// <returns>字符串表示</returns>
        public override string ToString()
        {
            return $"RpcResponse{{MessageId='{MessageId}', RequestId='{RequestId}', Success={IsSuccess()}, StatusCode={StatusCode}, Timestamp={Timestamp}}}";
        }
        
        /// <summary>
        /// 重写Equals方法
        /// </summary>
        /// <param name="obj">比较对象</param>
        /// <returns>是否相等</returns>
        public override bool Equals(object obj)
        {
            if (obj == null || GetType() != obj.GetType())
                return false;
            
            if (ReferenceEquals(this, obj))
                return true;
            
            var other = (RpcResponse)obj;
            
            return base.Equals(obj) &&
                   RequestId == other.RequestId &&
                   Equals(Result, other.Result) &&
                   Equals(Exception, other.Exception) &&
                   StatusCode == other.StatusCode &&
                   StatusMessage == other.StatusMessage;
        }
        
        /// <summary>
        /// 重写GetHashCode方法
        /// </summary>
        /// <returns>哈希码</returns>
        public override int GetHashCode()
        {
            int result = base.GetHashCode();
            result = 31 * result + (RequestId?.GetHashCode() ?? 0);
            result = 31 * result + (Result?.GetHashCode() ?? 0);
            result = 31 * result + (Exception?.GetHashCode() ?? 0);
            result = 31 * result + StatusCode;
            result = 31 * result + (StatusMessage?.GetHashCode() ?? 0);
            return result;
        }
        
        /// <summary>
        /// 创建RPC响应的构建器
        /// </summary>
        /// <returns>响应构建器</returns>
        public static Builder CreateBuilder()
        {
            return new Builder();
        }
        
        /// <summary>
        /// RPC响应构建器
        /// </summary>
        public class Builder
        {
            private string messageId;
            private string requestId;
            private object result;
            private Exception exception;
            private int statusCode = 200;
            private string statusMessage = "OK";
            
            public Builder SetMessageId(string messageId)
            {
                this.messageId = messageId;
                return this;
            }
            
            public Builder SetRequestId(string requestId)
            {
                this.requestId = requestId;
                return this;
            }
            
            public Builder SetResult(object result)
            {
                this.result = result;
                return this;
            }
            
            public Builder SetException(Exception exception)
            {
                this.exception = exception;
                if (exception != null)
                {
                    this.statusCode = 500;
                    this.statusMessage = "Internal Server Error";
                }
                return this;
            }
            
            public Builder SetStatusCode(int statusCode)
            {
                this.statusCode = statusCode;
                return this;
            }
            
            public Builder SetStatusMessage(string statusMessage)
            {
                this.statusMessage = statusMessage;
                return this;
            }
            
            public RpcResponse Build()
            {
                var response = new RpcResponse(messageId, requestId)
                {
                    Result = result,
                    Exception = exception,
                    StatusCode = statusCode,
                    StatusMessage = statusMessage
                };
                return response;
            }
        }
    }
} 